### PR TITLE
Add Conesnsus Exchange Types, and Move into NodeImplementation

### PR DESCRIPTION
**Pull request by bfish713**
Will close: https://github.com/EspressoSystems/HotShot/issues/927

### CODE REVIEW

File: src/tasks/mod.rs [hunk 1]
```diff
@@ -139,13 +141,45 @@ where
...
+    );
+    let _network_direct_task_handle = async_spawn(
+        network_task(
+            hotshot.clone(),
+            shut_down.clone(),
+            TransmitType::Direct,
+            exchange,
+        )
+        .instrument(info_span!("HotShot Direct Task",)),
+    );
```
Comment (shenkeyao) [inline src/tasks/mod.rs:164]: Just double-checking, do we plan to add these to the `HotShotInner` in the future? Not sure I understand why we add the DA tasks but not these two quorum tasks.

**Review by shenkeyao — APPROVED** [review]
Looks good! Only minor comments.

File: src/tasks/mod.rs (hunk 1 above)
Comment (bfish713) [reply src/tasks/mod.rs:164]: It should be in `TaskHandleInnder` I'll add it there.  It's used for shutdown so I will add it there now.

File: types/src/certificate.rs [hunk 2]
```diff
@@ -174,13 +177,11 @@ impl<TYPES: NodeType, LEAF: commit::Committable>
         self.signatures.clone()
     }
 
-    fn leaf_commitment(&self) -> Commitment<LEAF> {
-        // This function is only useful for QC. Will be removed after we have separated cert traits.
-        #[allow(deprecated)]
-        nll_todo()
+    fn leaf_commitment(&self) -> Commitment<TYPES::BlockType> {
+        self.block_commitment
```
[the remark this answers is not in the document]
Comment (bfish713) [reply types/src/certificate.rs:181]: Hmm I actually started using this in `is_valid_cert` to make sure the cert commitment is the same as the block commitment from the `DAProposal`

**Review by elliedavidson — APPROVED** [review]

File: src/traits/networking/centralized_web_server_network.rs [hunk 3]
```diff
@@ -46,36 +48,37 @@ use std::{
 };
 use surf_disco::error::ClientError;
 use tracing::{error, info};
-
 /// Represents the communication channel abstraction for the web server
 #[derive(Clone)]
 pub struct CentralizedWebCommChannel<
     TYPES: NodeType,
+    I: NodeImplementation<TYPES>,
```
[the remark this answers is not in the document]
Comment (bfish713) [reply src/traits/networking/centralized_web_server_network.rs:55]: Do you mean just here, or everywhere in the code?  It seems like `I: NodeImplementation` was used in a lot of places before this pull request

[the remark this answers is not in the document]
Comment (elliedavidson) [reply src/traits/networking/centralized_web_server_network.rs:55]: I meant everywhere in the code, though I had thought we changed `I` to something else earlier.  But I guess we didn't.  So no need to do it for this PR.  😁

[the remark this answers is not in the document]
Comment (bfish713) [reply src/traits/networking/centralized_web_server_network.rs:55]: It is `NODE` in one place, I think I'll close this PR and can open another with the rename

**Comment by shenkeyao** [comment]
Double-checking a minor thing: I noticed that there were many `error!` to `info!` changes. Were they intentional? Or maybe changed for the convenience of debugging but supposed to be changed back?

**Review by shenkeyao — APPROVED** [review]
LGTM!

**Comment by bfish713** [comment]
Apparently the new version of rust causes a new lint error which I couldn't see locally.  Updated rustc locally and fixed it, but now I need another review :(

**Review by elliedavidson — DISMISSED** [review]

**Review by elliedavidson — APPROVED** [review]

**Review by shenkeyao — APPROVED** [review]

**Pull request merged**